### PR TITLE
feat: add full program and exam content

### PR DESCRIPTION
**Pull request by zchsh**
> **Note**: 🏗 This PR targets the Certifications assembly branch. See #1447 for details.

## 🗒️ What

This PR implements content loading for the `exam` items on each individual certification program page.

## 🛠️ How

- Installs some `unified`-related utilities
- Sets up content utilities for working with FAQ content in MDX
    - `collect-md-content-by-heading` splits an MDX string into an array of `{ title, content }` items, where `title` is the heading of a section, and `content` is the MDX string for content within that section
    - `get-faqs-from-mdx` uses `collect-md-content-by-heading` to split an MDX document on `## Heading Two` headings, and also transforms the `content` string into `mdxSource` using `next-mdx-remote`
- Adds in `exams` to each Certification program's content schema
    - `exams` content will drive the exam details on each certifications page. It will also be used for the overview of certification programs on the landing page.
- Stubs in content for exams
    - Adds `exams` arrays to each `<program>.json` content file
    - Adds `.mdx` files for each `exam` item's FAQ content
- Adds a `prepare-page-content` utility for the individual program view
   - This uses the `get-faqs-from-mdx` utilities, and the updated schemas, to process MDX content for FAQs and pass it to the view component

## 🔗 Page Links

- [/certifications/infrastructure-automation][/certifications/infrastructure-automation]
- [/certifications/networking-automation][/certifications/networking-automation]
- [/certifications/security-automation][/certifications/security-automation]

## 🧪 Testing

- [ ] Visit each of the pages listed above.
    - Each page should now render an array of `exam` content in the content debug block
- [ ] Visit [/certifications][/certifications], it should still render as expected
    - No new content on the landing page, though

[preview]: https://dev-portal-git-zscertifications-program-content-hashicorp.vercel.app/
[/certifications]: https://dev-portal-git-zscertifications-program-content-hashicorp.vercel.app/certifications
[/certifications/infrastructure-automation]: https://dev-portal-git-zscertifications-program-content-hashicorp.vercel.app/certifications/infrastructure-automation
[/certifications/networking-automation]: https://dev-portal-git-zscertifications-program-content-hashicorp.vercel.app/certifications/networking-automation
[/certifications/security-automation]: https://dev-portal-git-zscertifications-program-content-hashicorp.vercel.app/certifications/security-automation

### CODE REVIEW

**Comment by zchsh** [comment]
@kendallstrautman This PR expands the content schema & processing for individual certification programs. We load in FAQ items from `.mdx` files within `src/content`.

This PR passes the prepared page content to the view, but doesn't go beyond that. Intent is to add components that render this content in the view as we expect in subsequent PRs 👍

File: src/content/certifications/exam-faqs/consul-associate-002.mdx [hunk 1]
```diff
@@ -0,0 +1,100 @@
+<!-- Each "## Heading Two" section in this document will be used to populate an FAQ item -->
```
Comment (kendallstrautman) [inline src/content/certifications/exam-faqs/consul-associate-002.mdx:1]: Interesting! Is this how it's set up on .com as well? I see now why we would want to add headings via html here with this setup.

Comment (kendallstrautman) [reply src/content/certifications/exam-faqs/consul-associate-002.mdx:1]: Not necessary for this batch of work but as a post-launch follow-up it'd be nice to have some authoring notes / guidelines. Perhaps a README at `content/certifications`?

File: src/views/certifications/content/utils/collect-md-content-by-heading.ts [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+/**
+ * A remark plugin that collects content nodes in the document,
+ * splitting them by headings
+ *
+ * Populates a provided collector variable with an array of sections.
+ * Each section consists of a title string, derived from the heading,
+ * as well as an array of top-level content nodes found under the heading.
+ */
+const contentSectionCollector: Plugin<[ContentSectionCollectorPluginOptions]> =
```
Comment (kendallstrautman) [inline src/views/certifications/content/utils/collect-md-content-by-heading.ts:46]: This is neat!

**Review by kendallstrautman — APPROVED** [review]
Very clean 💯

File: src/content/certifications/exam-faqs/consul-associate-002.mdx (hunk 1 above)
Comment (zchsh) [reply src/content/certifications/exam-faqs/consul-associate-002.mdx:1]: Sweet! I've just pushed up [content/certifications/README.md](https://github.com/hashicorp/dev-portal/blob/bd45acc953a260d56393323413b1248f85d4fbeb/src/content/certifications/README.md), and have linked to it from the main [content/README.md](https://github.com/hashicorp/dev-portal/blob/bd45acc953a260d56393323413b1248f85d4fbeb/src/content/README.md) in a new Certifications section.

**Pull request merged**